apluslms_roman/builder.py

```diff
@@ -20,16 +20,20 @@ def __init__(self, engine, config, observer=None):
 
 
     def get_steps(self, refs: list = None):
-        steps = [BuildStep.from_config(i, step) for i, step in enumerate(self.config.steps)]
         if refs:
-            name_dict = {step.name: step for step in steps}
-            step_list = [int(ref) if ref.isdigit() else ref for ref in refs]
-            try:
-                steps = [steps[ref] if isinstance(ref, int) else name_dict[ref] for ref in step_list]
-            except KeyError as e:
-                raise ValueError(_("No step named {}").format(e))
-            except IndexError as e:
-                raise ValueError(_("A step index was too big. Remember, indexing begins with 0."))
+            steps = []
+            for ref in refs:
+                try:
+                    steps.append(self.config.get_step(ref))
```

Comment:
What do you think if this would be written:

```
get_step = self.config.get_step
try:
    steps = [get_step(ref) for ref in refs]
except KeyError:
    raise ValueError(_("No step named %s") % format(ref))
except IndexError:
    raise IndexError(_(
        "Index %s is out of range. There are %d steps. Indexing begins at 0."
    ) % (ref, len(self.config.steps)))
```

Reply:
doesn't work like that, python doesn't remember ref outside of the list comprehension

Reply:
Damn, didn't notice `ref` in `raise ValueError(_("No step named %s") % format(ref))`. My bad.